### PR TITLE
[SPARK-9533] [PySpark] [ML] Add missing methods in Word2Vec ML

### PR DESCRIPTION
**Pull request by MechCoder**
After https://github.com/apache/spark/pull/7263 it is pretty straightforward to Python wrappers.

### CODE REVIEW

**Comment by MechCoder** [comment]
retest this please

File: mllib/src/main/scala/org/apache/spark/ml/feature/Word2Vec.scala [hunk 1]
```diff
@@ -153,7 +153,7 @@ class Word2VecModel private[ml] (
    * Returns a dataframe with two fields, "word" and "vector", with "word" being a String and
    * and the vector the DenseVector that it is mapped to.
    */
-  val getVectors: DataFrame = {
+  def getVectors: DataFrame = {
```
Comment (MechCoder) [inline mllib/src/main/scala/org/apache/spark/ml/feature/Word2Vec.scala]: @jkbradley this should be a def. Sorry that it went unnoticed by both of us.

Comment (jkbradley) [inline mllib/src/main/scala/org/apache/spark/ml/feature/Word2Vec.scala]: Actually, maybe both this and findSynonyms should be ```@transient lazy val```.

**Comment by jkbradley** [comment]
I'll make a pass now

**Comment by jkbradley** [comment]
Looks good except for maybe changing those Scala defs to transient lazy vals

**Comment by MechCoder** [comment]
@jkbradley Actally `findSynonyms` should be a method having parameters. Let me know if there is anything else.

**Comment by jkbradley** [comment]
LGTM, merging with master and branch-1.5
Thank you!

**Pull request closed without merging**